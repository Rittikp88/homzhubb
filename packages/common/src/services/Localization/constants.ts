// Add all languages you wish to support here
export enum supportedLanguages {
  English = 'en',
}
const whitelist = Object.values(supportedLanguages);

/**
 * Add translation files here, one per language
 * key must be from among the above enum `supportedLanguages`
 */
const resources = {
  en: require('@homzhub/common/src/assets/languages/en.json'),
};

// Add namespaces here
enum namespacesKey {
  common = 'common',
}
const namespaces = Object.values(namespacesKey);
const defaultNamespace = namespacesKey.common;

const fallback = 'en';

export const LocaleConstants = {
  whitelist,
  defaultNamespace,
  namespaces,
  fallback,
  resources,
};
