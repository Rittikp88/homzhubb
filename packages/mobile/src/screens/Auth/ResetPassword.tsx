import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { Formik, FormikProps, FormikValues } from 'formik';
import * as yup from 'yup';
import { FormUtils } from '@homzhub/common/src/utils/FormUtils';
import { AlertHelper } from '@homzhub/mobile/src/utils/AlertHelper';
import { UserService } from '@homzhub/common/src/services/UserService';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { FormButton, FormTextInput, DetailedHeader } from '@homzhub/common/src/components';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.ResetPassword>;

interface IResetPasswordState {
  password: string;
}

class ResetPassword extends Component<Props, IResetPasswordState> {
  public state = {
    password: '',
  };

  public render(): React.ReactNode {
    const { t } = this.props;
    const formData = { ...this.state };
    return (
      <View style={styles.container}>
        <DetailedHeader
          icon={icons.close}
          subTitle={t('auth:resetPassword')}
          subTitleType="large"
          subTitleColor={theme.colors.dark}
          onIconPress={this.handleIconPress}
          headerContainerStyle={styles.headerContainer}
        />
        <View style={styles.content}>
          <Formik onSubmit={this.onSubmit} validate={FormUtils.validate(this.formSchema)} initialValues={formData}>
            {(formProps: FormikProps<FormikValues>): React.ReactElement => (
              <>
                <FormTextInput
                  formProps={formProps}
                  inputType="password"
                  name="password"
                  label="Password"
                  helpText={t('auth:passwordValidation')}
                  placeholder={t('auth:enterPassword')}
                />
                <FormButton
                  formProps={formProps}
                  type="primary"
                  title={t('auth:resetPasswordButtonTitle')}
                  // @ts-ignore
                  onPress={formProps.handleSubmit}
                  containerStyle={styles.formButtonStyle}
                />
              </>
            )}
          </Formik>
        </View>
      </View>
    );
  }

  private onSubmit = (formProps: IResetPasswordState): void => {
    const { password } = formProps;
    const {
      navigation,
      route: {
        params: { token },
      },
    } = this.props;
    const payload = {
      action: 'SET_PASSWORD',
      payload: {
        token,
        password,
      },
    };
    UserService.resetPassword(payload)
      .then(() => {
        navigation.navigate(ScreensKeys.SuccessResetPassword);
        return null;
      })
      .catch((err) => {
        AlertHelper.error({ message: err });
      });
  };

  public handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.EmailLogin);
  };

  private formSchema = (): yup.ObjectSchema<{ password: string }> => {
    const { t } = this.props;
    return yup.object().shape({
      password: yup
        .string()
        .matches(FormUtils.passwordRegex, t('auth:passwordValidation'))
        .min(8, t('auth:minimumCharacters'))
        .required(t('auth:passwordRequired')),
    });
  };
}

export default withTranslation()(ResetPassword);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
  content: {
    paddingHorizontal: 24,
  },
  headerContainer: {
    borderBottomWidth: 0,
  },
  formButtonStyle: {
    flex: 0,
    marginVertical: 30,
  },
});
