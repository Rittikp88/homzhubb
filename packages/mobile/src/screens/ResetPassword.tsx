import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { Formik, FormikProps, FormikValues } from 'formik';
import * as yup from 'yup';
import { FormUtils } from '@homzhub/common/src/utils/FormUtils';
import { AlertHelper } from '@homzhub/mobile/src/utils/AlertHelper';
import { theme } from '@homzhub/common/src/styles/theme';
import { FormButton, FormTextInput, Header, Label } from '@homzhub/common/src/components';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.ResetPassword>;

interface IResetPasswordState {
  password: string;
}

class ResetPassword extends Component<Props, IResetPasswordState> {
  public state = {
    password: '',
  };

  public render(): React.ReactNode {
    const { t } = this.props;
    const formData = { ...this.state };
    return (
      <View style={styles.container}>
        <Header
          icon="close"
          subTitle={t('auth:resetPassword')}
          subTitleType="large"
          subTitleColor={theme.colors.dark}
          onIconPress={this.handleIconPress}
          headerContainerStyle={styles.headerContainer}
        />
        <View style={styles.content}>
          <Formik onSubmit={this.onSubmit} validate={FormUtils.validate(this.formSchema)} initialValues={formData}>
            {(formProps: FormikProps<FormikValues>): React.ReactElement => (
              <>
                <FormTextInput
                  formProps={formProps}
                  inputType="password"
                  name="password"
                  label="Password"
                  helpText={t('auth:passwordValidation')}
                  placeholder={t('auth:enterPassword')}
                />
                <FormButton
                  formProps={formProps}
                  type="primary"
                  title={t('auth:resetPasswordButtonTitle')}
                  onPress={formProps.handleSubmit}
                  containerStyle={styles.formButtonStyle}
                />
              </>
            )}
          </Formik>
          {/* TODO: Remove the use of Success Password Reset Label from here */}
          <Label
            type="large"
            textType="bold"
            style={styles.successResetPasswordLink}
            onPress={this.navigateToSuccessResetPassword}
          >
            Success Password Reset
          </Label>
        </View>
      </View>
    );
  }

  private onSubmit = (): void => {
    // TODO: Add the login on submit of Form
    AlertHelper.success({ message: 'Some message title' });
  };

  public handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.EmailLogin);
  };

  public navigateToSuccessResetPassword = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.SuccessResetPassword);
  };

  private formSchema = (): yup.ObjectSchema<{ password: string }> => {
    const { t } = this.props;
    return yup.object().shape({
      password: yup.string().required(t('auth:emailRequired')),
    });
  };
}

export default withTranslation()(ResetPassword);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
  content: {
    paddingHorizontal: 24,
  },
  headerContainer: {
    borderBottomWidth: 0,
  },
  formButtonStyle: {
    flex: 0,
    marginVertical: 30,
  },
  successResetPasswordLink: {
    color: theme.colors.blue,
    alignSelf: 'center',
  },
});
