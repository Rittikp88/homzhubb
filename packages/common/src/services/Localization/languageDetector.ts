import { Services, InitOptions, LanguageDetectorAsyncModule } from 'i18next';
import { supportedLanguages } from '@homzhub/common/src/services/Localization/constants';

const languageDetector: LanguageDetectorAsyncModule = {
  type: 'languageDetector',
  async: true,
  detect(callback: (lng: string) => void): void {
    callback(supportedLanguages.English);
  },
  init: (services: Services, detectorOptions: object, i18nextOptions: InitOptions): void => {},
  cacheUserLanguage: (lng: string): void => {},
};

export default languageDetector;
