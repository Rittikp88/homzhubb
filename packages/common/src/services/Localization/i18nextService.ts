import i18next, { i18n } from 'i18next';
import { initReactI18next } from 'react-i18next';
import { I18nManager } from 'react-native';
import languageDetector from '@homzhub/common/src/services/Localization/languageDetector';
import { LocaleConstants, supportedLanguages } from '@homzhub/common/src/services/Localization/constants';
import { Logger } from '@homzhub/common/src/utils/Logger';

class I18nextService {
  private readonly _instance: i18n;

  constructor() {
    this._instance = i18next.createInstance();
  }

  public init = async (): Promise<void> => {
    this._instance.use(initReactI18next).use(languageDetector);
    await this._instance.init(
      {
        whitelist: LocaleConstants.whitelist,
        resources: LocaleConstants.resources,
        fallbackLng: LocaleConstants.fallback,
        ns: LocaleConstants.namespaces,
        defaultNS: LocaleConstants.defaultNamespace,
        interpolation: {
          escapeValue: false,
        },
      },
      this.initCallback
    );
  };

  get instance(): i18n {
    return this._instance;
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  public t = (key: string, options?: any): string => {
    if (!this._instance) {
      this.init().then();
    }
    return this._instance.t(key, options);
  };

  public isRTL = (): boolean => {
    return I18nManager.isRTL;
  };

  public select<T>(map: { [position in 'rtl' | 'ltr']: T }): T {
    const key = this.isRTL() ? 'rtl' : 'ltr';
    return map[key];
  }

  public changeLanguage = async (lngCode: supportedLanguages): Promise<void> => {
    await this._instance.changeLanguage(lngCode);
  };

  private initCallback = (error: Error): void => {
    if (error) {
      Logger.error(error);
    } else {
      Logger.info('Loaded i18next');
    }
  };
}

const i18nService = new I18nextService();
export { i18nService as I18nService };
