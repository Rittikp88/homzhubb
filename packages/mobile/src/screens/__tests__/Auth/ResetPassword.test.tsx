import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { I18nextProvider } from 'react-i18next';
import { I18nService } from '@homzhub/common/src/services/Localization/i18nextService';
import ResetPassword from '@homzhub/mobile/src/screens/Auth/ResetPassword';

const mock = jest.fn();

describe('Reset Password Screen', () => {
  let component: ShallowWrapper;
  let props: any;

  beforeEach(async () => {
    props = {
      navigation: {
        navigate: mock,
      },
    };
    await I18nService.init();
    component = shallow(
      <I18nextProvider i18n={I18nService.instance}>
        <ResetPassword {...props} />
      </I18nextProvider>
    );
  });

  it('should render reset password screen', () => {
    expect(toJson(component.dive().dive())).toMatchSnapshot();
  });
});
