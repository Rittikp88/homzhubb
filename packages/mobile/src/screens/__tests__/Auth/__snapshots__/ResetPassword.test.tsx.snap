// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Screen should render reset password screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <DetailedHeader
    headerContainerStyle={
      Object {
        "borderBottomWidth": 0,
      }
    }
    icon="close"
    onIconPress={[Function]}
    subTitle="auth:resetPassword"
    subTitleColor="#001B36"
    subTitleType="large"
    testID="headerIconPress"
  />
  <View
    style={
      Object {
        "paddingHorizontal": 24,
      }
    }
  >
    <Formik
      initialValues={
        Object {
          "password": "",
        }
      }
      onSubmit={[Function]}
      validate={[Function]}
    >
      <Component />
    </Formik>
  </View>
</View>
`;
